src/klein/test/_strategies.py

```diff
@@ -26,10 +35,56 @@
 DrawCallable = Callable[[Callable[..., T]], T]
 
 
+if _PY3:
+    unichr = chr
+
+
+def idna_characters():  # pragma: no cover
+    # type: () -> str
+    """
+    Returns a string containing IDNA characters.
+    """
+    global _idnaCharacters
+
+    if _idnaCharacters is None:
+        result = []
+
+        # Data source "IDNA Derived Properties":
+        # https://www.iana.org/assignments/idna-tables-6.3.0/
+        #   idna-tables-6.3.0.xhtml#idna-tables-properties
+        dataFileName = join(dirname(__file__), "idna-tables-properties.csv")
```

Comment:
The right way to access data files is with `pkg_resources`. See [the docs](http://setuptools.readthedocs.io/en/latest/pkg_resources.html#basic-resource-access).

Reply:
`csv.reader` wants a text stream and `resource_stream` only provides a binary stream.  Not a problem! I'll wrap it in `TextIOWrapper`, which works in Py3 but not in Py2, because `file`, which is what you get from `resource_stream`, does not implement the `IOBase` interface for who the hell knows what reasons.

Reply:
I could read the whole file an shove it into a `TextIO`… but it's not a tiny file so that feels icky.